rust/list/src/lib.rs

```diff
@@ -346,10 +370,73 @@ pub unsafe extern "C" fn list_manager_get_all_labels(manager: *const ListManager
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn list_manager_create_item(manager: *mut ListManager, item: *const Item) {
+pub unsafe extern "C" fn list_manager_create_item(manager: *mut ListManager, name: *const c_char, due_date: *const time_t) {
+    let name = c_char_to_string(name);
+    log::d(&format!("Creating item: {:?}, {:?}, {:?}", name, due_date, manager)[..]);
+
     let manager = &mut*manager;
-    let item = &*item;
-    let _ = manager.create_item(&item);
+    let mut item = Item::default();
+
+    item.name = name;
+    item.due_date = Some(Timespec::new(due_date as i64, 0));
+
+    manager.create_item(&item);
+ 
+    // Let whoever's listening know that there are new items!
+    match CHANGED_CALLBACK {
+        Some(f) => f(),
+        None => ()
+    }
+}
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct ItemCSet {
```

Comment:
Naming nit: This is not a Set so shouldn't be called `ItemCSet`.

Also, see previous comment about why we need a wrapper around `[ItemC]`.

Reply:
Although, saying that, for iOS we need to attach array length information to the arrays we return via FFI. Currently we have a separate function that we call to get that information, but a struct that includes the array and length information would do the job more elegantly I think.